### PR TITLE
Support specifying server name, admin username, admin password and database username for MySQL

### PR DESCRIPTION
**Pull request by norshtein**
Implement the feature as the customer requests...

### CODE REVIEW

File: docs/modules/mysql.md [hunk 1]
```diff
@@ -34,6 +34,10 @@ Provisions a new MySQL DBMS and a new database upon it. The new database will be
 |----------------|------|-------------|----------|---------------|
 | `location` | `string` | The Azure region in which to provision applicable resources. | Y | |
 | `resourceGroup` | `string` | The (new or existing) resource group with which to associate new resources. | Y | |
+| `serverName` | `string` | Name of the PostgreSQL server. | N | A random generated string. |
```
Comment (zhongyi-zhang) [inline docs/modules/mysql.md]: typo: MySQL. Same below.

File: pkg/services/mysql/plan_schemas.go [hunk 2]
```diff
@@ -182,3 +227,90 @@ func firewallRuleValidator(
...
+		)
+	}
+	// For constraint3, it's really ambiguious which character is allowed.
+	// When tested on Azure Portal, character `!@$` is allowed,
+	// but character `#` is not allowed. I directly skip validating here
+	// and let MySQL RP return the error.
+	return nil
+}
+
+// passwordValidator validates postgreSQL password,
```
Comment (zhongyi-zhang) [inline pkg/services/mysql/plan_schemas.go]: typo in comment: MySQL

File: docs/modules/mysql.md (hunk 1 above)
Comment (norshtein) [reply docs/modules/mysql.md]: Nice cache, fixed.

File: pkg/services/mysql/plan_schemas.go (hunk 2 above)
Comment (norshtein) [reply pkg/services/mysql/plan_schemas.go]: fixed.

**Comment by norshtein** [comment]
@zhongyi-zhang  Seems API for text analytics has changed. Have temporarily disable `testCredentials` for text analytics service. Let's publish this release first.

**Pull request merged**